cli/src/commands/squash.rs

```diff
@@ -148,3 +153,79 @@ from the source will be moved into the parent.
     tx.finish(ui, format!("squash commit {}", commit.id().hex()))?;
     Ok(())
 }
+
+pub fn move_diff(
+    tx: &mut WorkspaceCommandTransaction,
+    settings: &UserSettings,
+    source: Commit,
+    mut destination: Commit,
+    matcher: Box<dyn Matcher>,
+    diff_selector: DiffSelector,
```

Comment:
nit: these arguments can be passed by reference, but it might be a bit weird if only `destination` is passed by value.

Reply:
I'll pass `matcher` and the `diff_selector` by reference